scipy/stats/biasedurn.pyx

```diff
@@ -29,10 +30,45 @@ cdef class _PyStochasticLib3:
     cdef StochasticLib3 c_sl3
 
     def __cinit__(self, int seed):
-        self.c_sl3 = StochasticLib3(seed)
-                
+        self.c_sl3 = StochasticLib3(seed);
+        
+    def FillCache(self, double[::1] rand_cache):
+        self.c_sl3.FillCache(&rand_cache[0], rand_cache.shape[0]);
+        
+    def Random(self):
+        x = self.c_sl3.Random();
+        return x;
+        
     def SetAccuracy(self, double accur):
         return self.c_sl3.SetAccuracy(accur)
-
+        
+    def rvs_fisher(self, int n, int m, int N, double odds, int size, int FOS=1):
+        
+        rvs = np.zeros(size)
+        
+        if (n < 30 and N < 1024 and odds > 1.E-5 and odds < 1.E5):
+            # inversion by chop down method will be used
+            cache = np.random.rand(size)
+            self.FillCache(cache)
+            for i in range(size):
+                rvs[i] = self.FishersNCHyp(n, m, N, odds)
+        else:
+            # ratio-of-uniforms method will be used
+            # size of random cache needed is indeterminate
+            i = 0
+            remaining_size = size
+            while i < size:
+                try:
+                    cache = np.random.rand(remaining_size * FOS)
+                    self.FillCache(cache)
+                    for j in range(i, size):
+                        rvs[j] = self.FishersNCHyp(n, m, N, odds)
+                        i += 1
+                except IndexError:
+                    remaining_size = size - i
+                    FOS *= 2
```

Comment:
`FOS` (Factor of Safety) is a multiple on the size of the random cache. This increases every time the cache runs out to be more adaptive - in case there are parameters for which the algorithm burns through a lot of random numbers.